ANTLR/src/Compiler/CodeGeneration/JavaScriptCodeGenerationVisitor.java

```diff
@@ -43,29 +43,49 @@ public String visitCodeStmt(CodeStmtContext ctx) {
         return visit(ctx.stmt());
     }
 
+
     /**
-     * Generates code for a function definition node.
+     * Generates code for an if function definition node.
      * @param ctx The tree node in question.
      * @return A string of the form: {@code "function id(param0, param1 , ... , paramN) &#123; functionBody &#125;}".
      */
     @Override
-    public String visitFuncDef(FuncDefContext ctx) {
+    public String visitIfFunction(IfFunctionContext ctx) {
+        String result = initiateFuncDef(ctx.typeAndId(), ctx.funcDefParams());
+
+        int stmtsSize =  ctx.getRuleContexts(StmtsContext.class).size();
+
+        result += getStringFromTokenList(i -> visit(ctx.stmts(i)), 0, stmtsSize, " ");
```

Comment:
Pretty sure this method is overloaded, so that you can reduce it to this:
```suggestion
        result += getStringFromTokenList(i -> visit(ctx.stmts(i)), stmtsSize);
```

Reply:
Det var den ikke, så det lavede jeg